src/components/PercentageForm.tsx

```diff
@@ -0,0 +1,101 @@
+import type {ForwardedRef} from 'react';
+import React, {forwardRef, useCallback, useMemo, useRef, useState} from 'react';
+import type {NativeSyntheticEvent, TextInputSelectionChangeEventData} from 'react-native';
+import useLocalize from '@hooks/useLocalize';
+import * as MoneyRequestUtils from '@libs/MoneyRequestUtils';
+import CONST from '@src/CONST';
+import TextInput from './TextInput';
+import type {BaseTextInputRef} from './TextInput/BaseTextInput/types';
+
+type PercentageFormProps = {
+    /** Amount supplied by the FormProvider */
+    value?: string;
+
+    /** Error to display at the bottom of the component */
+    errorText?: string;
+
+    /** Callback to update the amount in the FormProvider */
+    onInputChange?: (value: string) => void;
+
+    label?: string;
+};
+
+/**
+ * Returns the new selection object based on the updated amount's length
+ */
+const getNewSelection = (oldSelection: {start: number; end: number}, prevLength: number, newLength: number) => {
+    const cursorPosition = oldSelection.end + (newLength - prevLength);
+    return {start: cursorPosition, end: cursorPosition};
+};
+
+function PercentageForm({value: amount, errorText, onInputChange, label, ...rest}: PercentageFormProps, forwardedRef: ForwardedRef<BaseTextInputRef>) {
+    const {toLocaleDigit, numberFormat} = useLocalize();
+
+    const textInput = useRef<BaseTextInputRef | null>(null);
+
+    const currentAmount = useMemo(() => (typeof amount === 'string' ? amount : ''), [amount]);
+
+    const [selection, setSelection] = useState({
+        start: currentAmount.length,
+        end: currentAmount.length,
+    });
+
+    const forwardDeletePressedRef = useRef(false);
+
+    /**
+     * Sets the selection and the amount accordingly to the value passed to the input
+     * @param newAmount - Changed amount from user input
+     */
+    const setNewAmount = useCallback(
+        (newAmount: string) => {
+            // Remove spaces from the newAmount value because Safari on iOS adds spaces when pasting a copied value
+            // More info: https://github.com/Expensify/App/issues/16974
+            const newAmountWithoutSpaces = MoneyRequestUtils.stripSpacesFromAmount(newAmount);
```

Comment:
I agree, but let's move it to a separate ticket. I think those forms should be connected to some extent (like exporting to a shared part as a base), but I wanted to avoid unnecessary conflicts.

Reply:
Sounds good!

Reply:
Resolving for now but moved to [this issue](https://github.com/Expensify/App/issues/48499)